fedn/fedn/network/combiner/aggregators/fedopt.py

```diff
@@ -54,24 +49,55 @@ def combine_models(self, helper=None, delete_models=True, params=None):
         :type max_nr_models: int, optional
         :param delete_models: Delete models from storage after aggregation, defaults to True
         :type delete_models: bool, optional
-        :param params: Additional key-word arguments.
-        :type params: dict
+        :param parameters: Aggregator hyperparameters.
+        :type parameters: `fedn.utils.parmeters.Parameters`, optional
         :return: The global model and metadata
         :rtype: tuple
         """
 
-        params = ast.literal_eval(params)
         data = {}
         data['time_model_load'] = 0.0
         data['time_model_aggregation'] = 0.0
 
-        # Override default hyperparameters:
-        if params:
-            for key, value in self.default_params.items():
-                if key not in params:
-                    params[key] = value
+        # Define parameter schema
+        parameter_schema = {
```

Comment:
I would have this a required attribute of AggregatorBase, even if it can be empty